File: android/app/src/main/java/com/dawainnovation/lens/NativeHapticsPlugin.kt
package com.dawainnovation.lens

import android.content.Context
import android.os.Build
import android.os.VibrationEffect
import android.os.Vibrator
import android.os.VibratorManager
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "NativeHaptics")
class NativeHapticsPlugin : Plugin() {

    private val vibrator: Vibrator by lazy {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
            val vibratorManager = context.getSystemService(Context.VIBRATOR_MANAGER_SERVICE) as VibratorManager
            vibratorManager.defaultVibrator
        } else {
            @Suppress("DEPRECATION")
            context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
        }
    }

    @PluginMethod
    fun impact(call: PluginCall) {
        val style = call.getString("style") ?: "medium"
        
        val effect = when (style) {
            "light" -> VibrationEffect.createPredefined(VibrationEffect.EFFECT_TICK)
            "heavy" -> VibrationEffect.createPredefined(VibrationEffect.EFFECT_CLICK)
            else -> VibrationEffect.createPredefined(VibrationEffect.EFFECT_CLICK)
        }
        
        vibrator.vibrate(effect)
        call.resolve()
    }

    @PluginMethod
    fun notification(call: PluginCall) {
        val type = call.getString("type") ?: "success"
        
        when (type) {
            "success" -> {
                val timings = longArrayOf(0, 40, 40, 40)
                val amplitudes = intArrayOf(0, 100, 0, 150)
                vibrator.vibrate(VibrationEffect.createWaveform(timings, amplitudes, -1))
            }
            "warning" -> {
                val timings = longArrayOf(0, 100, 40, 100)
                val amplitudes = intArrayOf(0, 200, 0, 200)
                vibrator.vibrate(VibrationEffect.createWaveform(timings, amplitudes, -1))
            }
            "error" -> {
                val timings = longArrayOf(0, 50, 40, 50, 40, 50, 40, 150)
                val amplitudes = intArrayOf(0, 150, 0, 150, 0, 150, 0, 255)
                vibrator.vibrate(VibrationEffect.createWaveform(timings, amplitudes, -1))
            }
        }
        call.resolve()
    }
}

File: ios/App/App/NativeHapticsPlugin.swift
import Foundation
import Capacitor

@objc(NativeHapticsPlugin)
public class NativeHapticsPlugin: CAPPlugin {

    @objc func impact(_ call: CAPPluginCall) {
        let styleStr = call.getString("style") ?? "medium"
        var style: UIImpactFeedbackGenerator.FeedbackStyle = .medium

        switch styleStr {
        case "light":
            style = .light
        case "heavy":
            style = .heavy
        default:
            style = .medium
        }

        DispatchQueue.main.async {
            let generator = UIImpactFeedbackGenerator(style: style)
            generator.prepare()
            generator.impactOccurred()
        }
        call.resolve()
    }

    @objc func notification(_ call: CAPPluginCall) {
        let typeStr = call.getString("type") ?? "success"
        var type: UINotificationFeedbackGenerator.FeedbackType = .success

        switch typeStr {
        case "warning":
            type = .warning
        case "error":
            type = .error
        default:
            type = .success
        }

        DispatchQueue.main.async {
            let generator = UINotificationFeedbackGenerator()
            generator.prepare()
            generator.notificationOccurred(type)
        }
        call.resolve()
    }
}
